### PR TITLE
Import mlwh query functions from vector-ops

### PR DESCRIPTION
**Pull request by amakunin**
This PR imports mlwh querying functions from https://github.com/malariagen/vector-ops/pull/633. Two queries possible:
- fetch sequencing metadata given sample IDs
- fetch three types of sample IDs given one type of sample IDs

@alimanfoo @tnguyensanger can you review this?

### CODE REVIEW

File: scripts/query_mlwh.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    Parameters:
+    db_config_file - path to mlwh config
+    sample_list - list of sample IDs, 
+    id_type - type of sample ID, valid values: 
+        - 'name' (corresponds to 'sanger_sample_id')
+        - 'supplier_name' (sample name in SequenceScape manifest, sometimes corresponds to 'derived_sample_id')
+        - 'accession_number' (corresponds to 'ebi_sample_acc')
+    
+    Return pandas dataframe of sample metadata
+    '''
```
Comment (alimanfoo) [inline scripts/query_mlwh.py]: No objection to merging this as-is if you are pressed for time, but FWIW I like to follow the [numpy docstring standard](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard) for docstrings.

File: scripts/query_mlwh.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                     "library",
+                     "manual_qc", 
+                     "run_complete", 
+                     "qc_complete",
+                     "run_id", 
+                     "lane_index", 
+                     "tag_index"])
+        
+        # include empty ebi run accession column (data not available from mlwh)
+        sql_data["ena_run_accession"] = ""
```
Comment (alimanfoo) [inline scripts/query_mlwh.py]: Can we just drop this line?

File: scripts/query_mlwh.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        
+        return (sql_data)
+
+    finally:           
+        if db and db.open:
```
Comment (alimanfoo) [inline scripts/query_mlwh.py]: Small point, but this check seems unnecessary. I.e., you can't get to this point unless `db` is an open connection. 

Also usually one calls `cursor.close()` before `db.close()`, but maybe that's not needed as `db.close()` will handle all cleanup.

File: scripts/query_mlwh.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                                  sample_list='", "'.join(sample_list))
+        cursor.execute(query)
+
+        # Fetch tuple of tuples.  Each inner tuple is a row.
+        db_results = cursor.fetchall()
+        
+        sql_data = pd.DataFrame.from_records(
+            list(db_results),
+            columns=["seqscape_sample_name", 
+                     "seqscape_sample_supplier_name",
```
Comment (alimanfoo) [inline scripts/query_mlwh.py]: Suggest that column names in the returned dataframe are made consistent between this function and the previous function.

**Review by alimanfoo — COMMENTED** [review]
Thanks Alex, a couple of small comments.

**Comment by alimanfoo** [comment]
I did some more work on this module in https://github.com/malariagen/vector-ops/pull/633 which includes changes addressing the comments I made here. I suggest we get that merged then revisit this with the new code from vector-ops.

**Comment by amakunin** [comment]
Code updated from https://github.com/malariagen/vector-ops/pull/633. Ready for review again.

**Review by bstlaure — COMMENTED** [review]
@amakunin, @alimanfoo -
this script works fine and is needed to run build-fofn-mlwh.ipynb

Can we merge this to the master of vector-tools?

**Pull request merged**